app/services/record_declarations/base.rb

```diff
@@ -18,7 +20,7 @@ class Base
     validate :date_has_the_right_format
 
     validates :declaration_type, inclusion: { in: :valid_declaration_types, message: I18n.t(:invalid_declaration_type) }
-    delegate :schedule, :participant_declarations, :fundable?, to: :user_profile, allow_nil: true
```

Comment:
remove an extra later of indirection with reduces the cognitive load needed to understand this class